components/count-badge/count-badge.js

```diff
@@ -172,19 +226,37 @@ class CountBadge extends RtlMixin(LitElement) {
 		if (this.maxDigits && this.number.toString().length > this.maxDigits) {
 			numberString = `${'9'.repeat(this.maxDigits)}+`;
 		}
+		const iconSizeStyle = {};
+		if (this.icon && this.iconSize) {
+			iconSizeStyle.height = `${this.iconSize}px`;
+			iconSizeStyle.width = `${this.iconSize}px`;
+		}
 		return html`
-        	<div
-			class="d2l-count-badge-number" 
-			id="${this._badgeId}"
+			<div 
+			class="d2l-count-badge-wrapper"
+			@blur="${this._onBlur}"
+			@focus="${this._onFocus}"
+			id="${ifDefined(this.icon ? undefined : this._badgeId)}"
 			tabindex="${ifDefined(this.tabStop || this.hasTooltip ? '0' : undefined)}" 
 			aria-labelledby="${ifDefined(this.hasTooltip ? undefined : this._textId)}">
-					<div aria-hidden="true">${numberString}</div>
-					${this.hasTooltip ?
-		html`<d2l-tooltip id="${this._textId}" for="${this._badgeId}" aria-live="${this.announceChanges ? 'polite' : 'off'}">${this.text}</d2l-tooltip>`
-		: html`<span id="${this._textId}" class="d2l-offscreen" aria-live="${this.announceChanges ? 'polite' : 'off'}">"${this.text}"</span>`}
-			</div>
+				<div class="d2l-count-badge-number">
+						<div aria-hidden="true">${numberString}</div>		
+				</div>
+				${this.icon ? html`<d2l-icon role="presentation" id="${this._badgeId}" icon="${ifDefined(this.icon)}" class="d2l-button-icon" style=${styleMap(iconSizeStyle)}></d2l-icon></div>` : null }
+				${this.hasTooltip ?
+		html`<d2l-tooltip id="${this._textId}" ?force-show="${this._forceTooltipOn}" aria-live="${this.announceChanges ? 'polite' : 'off'}" for="${this._badgeId}">${this.text}</d2l-tooltip>`
+		: html`<span id="${this._textId}" aria-live="${this.announceChanges ? 'polite' : 'off'}" class="d2l-offscreen">"${this.text}"</span>`}
+			<div>
 			`;
 	}
+
+	_onBlur() {
+		this._forceTooltipOn = false;
```

Comment:
Forcing the tooltip on is required when the icon property is set, as we want the tooltip to be positioned on the icon rather than the number or the whole badge. The tooltip's for property then gets set to the icon, meaning it only shows up when hovering on the icon directly. Forcing the tooltip on focus ensures the user can hover anywhere and focus on the badge and still get the tooltip.

Reply:
Yikes... that's complicated, and I think is going to mess up the tooltip's automatic adding of `aria-labelledby`... since the icon itself is presentational.

Would it be worth considering what splitting this into 2 components might look like? i.e. `<d2l-count-badge>` and `<d2l-count-badge-icon>`? Seems like we have a bunch of things that shift significantly when there's an icon and certain attribute/behaviours that get forced on. Lots of CSS that's icon-specific as well.